### PR TITLE
Upgrade web5/api to 0.9.4

### PR DESCRIPTION
**Pull request by leordev**
Requested at:
- #1437

### CODE REVIEW

**Comment by angiejones** [comment]
@leordev ready for PR to go in review?

**Comment by leordev** [comment]
> @leordev ready for PR to go in review?

Should we wait for the instructions to be added? Or will that go in a new PR?

**Comment by angiejones** [comment]
> > @leordev ready for PR to go in review?
> 
> @angiejones Should we wait for the instructions to be added? Or will that go in a new PR? https://discord.com/channels/937858703112155166/1245124090734317749/1245174191909765191

Ok great. @blackgirlbytes wants to add some docs here

File: site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-web-nodes/delete-from-dwn.test.js [hunk 1]
```diff
@@ -20,4 +20,61 @@ describe('delete-from-dwn', () => {
...
+    // :snippet-start: pruneRecords
+    const { status: deleteStatus } = await web5.dwn.records.delete({
+      message: {
+        recordId: parentRecord.id,
+        //highlight-next-line
+        prune: true
+      }
+    });
+    // :snippet-end:
+    expect(deleteStatus.code).toBe(202);
```
[the remark this answers is not in the document]
Comment (blackgirlbytes) [reply site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-web-nodes/delete-from-dwn.test.js]: okay added!

File: site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-web-nodes/delete-from-dwn.test.js [hunk 2]
```diff
@@ -20,4 +20,80 @@ describe('delete-from-dwn', () => {
...
+    const { records: childrenRecordsAfterDelete } = await web5.dwn.records.query({
+      message: {
+        filter: {
+          protocol: protocol.definition.protocol,
+          protocolPath: 'post/comment'
+        }
+      }
+    });
+
+    const { records: parentRecordsAfterDelete } = await web5.dwn.records.query({
```
Comment (blackgirlbytes) [inline site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-web-nodes/delete-from-dwn.test.js:87]: it seems i couldnt use parentRecord and childrenRecord. Instead..I had to query again to see if the records existed. Feel like this is an unexpected devex. 🤔

**Review by angiejones — APPROVED** [review]

**Review by blackgirlbytes — APPROVED** [review]
let's ship it!

**Pull request merged**